addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCAnalogBlockHandler.java

```diff
@@ -0,0 +1,282 @@
+
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.openhab.binding.plclogo.PLCLogoBindingConstants;
+import org.openhab.binding.plclogo.config.PLCLogoAnalogConfiguration;
+import org.openhab.binding.plclogo.internal.PLCLogoDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import Moka7.S7;
+
+/**
+ * The {@link PLCAnalogBlockHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexander Falkenstern - Initial contribution
+ */
+public class PLCAnalogBlockHandler extends PLCBlockHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(PLCAnalogBlockHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_ANALOG);
+
+    private PLCLogoAnalogConfiguration config = getConfigAs(PLCLogoAnalogConfiguration.class);
+    private long oldValue = Long.MAX_VALUE;
+
+    /**
+     * {@inheritDoc}
+     */
+    public PLCAnalogBlockHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        final Thing thing = getThing();
+        Objects.requireNonNull(thing, "PLCAnalogBlockHandler: Thing may not be null.");
+
+        final Bridge bridge = getBridge();
+        Objects.requireNonNull(bridge, "PLCAnalogBlockHandler: Bridge may not be null.");
+
+        synchronized (config) {
+            config = getConfigAs(PLCLogoAnalogConfiguration.class);
+        }
+
+        final String name = config.getBlockName();
+        logger.debug("Initialize LOGO! {} analog handler.", name);
+
+        scheduler.execute(new Runnable() {
+            @Override
+            public void run() {
+                if (config.isBlockValid()) {
+                    ThingBuilder tBuilder = editThing();
+
+                    String text = config.isInputBlock() ? INPUT_CHANNEL : OUTPUT_CHANNEL;
+                    text = text.substring(0, 1).toUpperCase() + text.substring(1);
+                    tBuilder = tBuilder.withLabel(bridge.getLabel() + ": " + text + " " + name);
```

Comment:
Sorry that I did not reply earlier, I was on hoiday

No builders are general thing, but always return themselves so you easily chain method invocations leading to better looking code and since it return itself (ie also mean it changes the state of the builder) you can either ignore the return value or chain them but to overwrite the builder with a response is a little strange.